### DISCUSSION TITLE
Cant get TOC to work?

### DISCUSSION THREAD

**Discussion by nekosophia** [GetPublii/Publii#1633] [Help]
I've only used the table of contents once before, and it was on a fairly short file but it seemed to work fine. For some reason, it wants to fight me in the current file and I cant for the life of me figure out why. My structure is -

H2
H3
H3
H2
H3
...etc

But for some reason, it's showing an empty 1 slot, when the only text prior to 'Introduction' is paragraph text above the 'read more'. Any ideas? I'm using simple theme, and I've only overridden main.css, header.hbs, and post.hbs.

![2wa5iw7wTA](https://github.com/user-attachments/assets/926d3181-5da7-4b2e-8436-2938945699ae)

Edit: I took a look at the html in the page, and for some reason a H2 had been generated at the top of the page, with no text, so wasn't showing in the WYSIWYG editor. removed that and it worked perfectly

**Comment by miditaur** [comment]
Everything is very simple here! The content is provided by the TinyMCE editor and the cause of your problem is related to it. Sometimes it  that an empty `<h2>` line is generated, which simply needs to be removed. Open your publication in html - click the corresponding button in the toolbar and you will see a generated line like - 
`<h2 id="mcetoc_1ikr5hpf930"></h2>` , just delete it and apply the changes!
**Don't forget to update the TOC!**
